File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
      # Bottom Up approach - Tabulation
      pascals = [[0]*(i+1) for i in range(rowIndex+1)]
      
      # Each row has a 1 in the beginning and end
      for i in range(len(pascals)):
        pascals[i][0] = pascals[i][-1] = 1
     
      for r in range(2, len(pascals)):        
        for c in range(1, r):
          prev = pascals[r-1]
          pascals[r][c] = prev[c-1] + prev[c]
      
      return pascals[rowIndex]
      

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
  // Initialize rows with empty arrays
  const pascals = new Array(rowIndex + 1).fill([]);
  for (let row = 0; row < pascals.length; row++) {
    pascals[row] = new Array(row + 1).fill(0);
  }
  pascals[0] = [1];
  pascals[1] = [1, 1];
  
  for (let i = 2; i < pascals.length; i++) {
    // Each row always has 1 in the beginning and end
    pascals[i][0] = 1;
    pascals[i][pascals[i].length-1] = 1;
    
    for (let j = 1; j < pascals[i].length-1; j++) {
      const prevRow = pascals[i-1];
      pascals[i][j] = prevRow[j-1] + prevRow[j];
    }
  }
  
  return pascals[rowIndex];
};